src/compiler/scala/tools/nsc/transform/Mixin.scala

```diff
@@ -266,11 +261,23 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
               //     level.
 
               lazy val mixinSuperInterfaces = mixinClass.ancestors.filter(_.isTraitOrInterface)
-              val needsForwarder = competingMethods.exists(m => {
-                !m.owner.isTraitOrInterface ||
-                  (!m.isDeferred && !mixinSuperInterfaces.contains(m.owner))
-              })
-              if (needsForwarder)
+
+              @tailrec
+              def existsCompetingMethod(baseClasses: List[Symbol]): Boolean = baseClasses match {
+                case baseClass :: rest =>
+                  if (baseClass ne mixinClass) {
+                    val m = member.overriddenSymbol(baseClass)
+                    val isCompeting = m.exists && {
+                      !m.owner.isTraitOrInterface ||
+                        (!m.isDeferred && !mixinSuperInterfaces.contains(m.owner))
```

Comment:
I had a failure at some point because `isNonBottomSubclass` includes parent classes (not only traits/interfaces). I can't remember the details, so I pushed a commit here and trying it out. It looks like class members are handled in the line above (`m.owner.isTraitOrInterface`) so it shouldn't make a difference.